haystack/nodes/retriever/dense.py

```diff
@@ -1072,11 +1072,18 @@ def retrieve_batch(
         if scale_score is None:
             scale_score = self.scale_score
 
-        query_embs: List[np.ndarray] = []
-        for batch in self._get_batches(queries=queries, batch_size=batch_size):
-            query_embs.extend(self.embed_queries(queries=batch))
+        # embed_queries is already batched within by batch_size
+        query_embs: np.ndarray = self.embed_queries(queries=queries)
+        batched_query_embs: List[np.ndarray] = []
+        for i in range(0, len(query_embs), batch_size):
+            batched_query_embs.extend(query_embs[i : i + batch_size])
```

Comment:
It seems to me like this loop is not doing anything.  Have a look below:

```python
query_embs = [1, 2, 3, 4, 5, 6, 7, 8, 9]
batched_query_embs = []
batch_size = 3

for i in range(0, len(query_embs), batch_size):
    batched_query_embs.extend(query_embs[i : i + batch_size])

assert query_embs == batched_query_embs
```

Reply:
Ahh good catch! Sorry, I meant to use `.append` since I want to recreate the batches before sending it off to `document_store.query_by_embedding_batch(`.

Reply:
Actually I double checked and `.extend` also works here. If we update your example to use 
```python
query_embs = np.array([[1, 2, 3, 4, 5, 6, 7, 8, 9], [1, 2, 3, 4, 5, 6, 7, 8, 9]])
batched_query_embs = []
batch_size = 1
...
assert query_embs != batched_query_embs
```
this is because we are extending or appending numpy arrays into a python list which is handled differently than if `query_embs` was already a python list.